Assets/BossRoom/Scripts/Server/Net/ServerSessionManager.cs

```diff
@@ -0,0 +1,272 @@
+        /// See GNH_Client.StartClient for the complementary logic that runs when the client starts its connection.
+        /// </summary>
+        /// <remarks>
+        /// Since our game doesn't have to interact with some third party authentication service to validate the identity of the new connection, our ApprovalCheck
+        /// method is simple, and runs synchronously, invoking "callback" to signal approval at the end of the method. MLAPI currently doesn't support the ability
+        /// to send back more than a "true/false", which means we have to work a little harder to provide a useful error return to the client. To do that, we invoke a
+        /// client RPC in the same channel that MLAPI uses for its connection callback. Since that channel ("MLAPI_INTERNAL") is both reliable and sequenced, we can be
+        /// confident that our login result message will execute before any disconnect message.
+        /// </remarks>
+        /// <param name="connectionData">binary data passed into StartClient. In our case this is the client's GUID, which is a unique identifier for their install of the game that persists across app restarts. </param>
+        /// <param name="clientId">This is the clientId that MLAPI assigned us on login. It does not persist across multiple logins from the same client. </param>
+        /// <param name="callback">The delegate we must invoke to signal that the connection was approved or not. </param>
+        private void ApprovalCheck(byte[] connectionData, ulong clientId, MLAPI.NetworkManager.ConnectionApprovedDelegate callback)
+        {
+            if (connectionData.Length > k_MaxConnectPayload)
+            {
+                return;
+            }
+
+            string payload = System.Text.Encoding.UTF8.GetString(connectionData);
+            var connectionPayload = JsonUtility.FromJson<ConnectionPayload>(payload); // https://docs.unity3d.com/2020.2/Documentation/Manual/JSONSerialization.html
+            if (connectionPayload == null)
+            {
+                return;
+            }
+            
+            Debug.Log("Host ApprovalCheck: connecting client GUID: " + connectionPayload.clientGUID);
+
+            ConnectStatus gameReturnStatus = ConnectStatus.Success;
+
+            //Test for Duplicate Login.
+            if (m_ClientData.ContainsKey(connectionPayload.clientGUID))
+            {
+                if (Debug.isDebugBuild)
+                {
+                    Debug.Log($"Client GUID {connectionPayload.clientGUID} already exists. Because this is a debug build, we will still accept the connection");
+                    while (m_ClientData.ContainsKey(connectionPayload.clientGUID)) { connectionPayload.clientGUID += "_Secondary"; }
+                }
+                else
+                {
+                    gameReturnStatus = ConnectStatus.LoggedInAgain;
```

Comment:
beside for checking for success, looks like we're not using that status for anything else?